kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -251,27 +240,47 @@ def exec_prove(
         haskell_args += ['--bug-report', str(spec_file.with_suffix(''))]
     if depth is not None:
         prove_args += ['--depth', str(depth)]
+    if claims:
+        prove_args += ['--claims', ','.join(claims)]
+    if exclude_claims:
+        prove_args += ['--exclude', ','.join(exclude_claims)]
     final_state = kevm.prove(spec_file, spec_module_name=spec_module, args=prove_args, haskell_args=haskell_args)
     print(kevm.pretty_print(final_state) + '\n')
+    if not (type(final_state) is KApply and final_state.label.name == '#Top'):
+        _LOGGER.error('Proof failed!')
+        sys.exit(1)
 
 
 def exec_foundry_prove(
     profile: bool,
     foundry_out: Path,
-    contract: Optional[str],
     includes: List[str],
     debug_equations: List[str],
     bug_report: bool,
-    spec_module: Optional[str],
     depth: Optional[int],
+    tests: Iterable[str] = (),
+    exclude_tests: Iterable[str] = (),
     **kwargs,
 ) -> None:
     _ignore_arg(kwargs, 'main_module', f'--main-module: {kwargs["main_module"]}')
     _ignore_arg(kwargs, 'syntax_module', f'--syntax-module: {kwargs["syntax_module"]}')
     _ignore_arg(kwargs, 'definition_dir', f'--definition: {kwargs["definition_dir"]}')
+    _ignore_arg(kwargs, 'spec_module', f'--spec-module: {kwargs["spec_module"]}')
     definition_dir = foundry_out / 'kompiled'
     spec_file = definition_dir / 'foundry.k'
-    spec_module = contract.upper() + '-BIN-RUNTIME-SPEC' if contract else 'FOUNDRY-SPEC'
+    spec_module = 'FOUNDRY-SPEC'
+    claims = []
+    exclude_claims = []
+    for _t in tests:
+        _m, _c = _t.split('.')
+        _m = _m.upper() + '-BIN-RUNTIME-SPEC'
+        _c = _c.replace('_', '-')
+        claims.append(f'{_m}.{_c}')
+    for _t in exclude_tests:
+        _m, _c = _t.split('.')
+        _m = _m.upper() + '-BIN-RUNTIME-SPEC'
+        _c = _c.replace('_', '-')
+        exclude_claims.append(f'{_m}.{_c}')
```

Comment:
Do we expect more conversions from Foundry-names to KEVM-names like this? Then it would be nice to extract this logic into a separate function.

Reply:
I have done this one.